KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift

```diff
@@ -540,10 +540,21 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
     }
 
     private func reportBookingEvent() {
-        guard let trip = trip else {
+        guard let origin = journeyDetails.originLocationDetails
+        else {
             return
         }
-        analytics.bookingRequested(tripDetails: trip)
+        
+        if trip == nil {
+            trip = TripInfo(
+                origin: origin.toTripLocationDetails(),
+                destination: journeyDetails.destinationLocationDetails?.toTripLocationDetails(),
+                dateScheduled: journeyDetails.scheduledDate,
+                quote: quote.toTripQuote()
+            )
+        }
+        
+        analytics.bookingRequested(tripDetails: trip!)
```

Comment:
To avoid force-unwrapping you could change the code a bit:
```
(nested) func buildTripForAnalytics() -> TripInfo { 
     TripInfo(
                origin: origin.toTripLocationDetails(),
                destination: journeyDetails.destinationLocationDetails?.toTripLocationDetails(),
                dateScheduled: journeyDetails.scheduledDate,
                quote: quote.toTripQuote()
            )
}

analytics.bookingRequested(tripDetails: trip ?? buildTripForAnalytics)
```